src/common/snippets/include/snippets/runtime_optimizer.hpp

```diff
@@ -0,0 +1,31 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "snippets/lowered/linear_ir.hpp"
+#include "snippets/lowered/pass/pass.hpp"
+#include "snippets/runtime_configurator.hpp"
+
+namespace ov {
+namespace snippets {
+namespace lowered {
+namespace pass {
+/**
+ * @class RuntimeOptimizer
+ * @brief Base class for runtime optimizers that operate on LinearIR and RuntimeConfigurator during
+ * RuntimeConfigurator::update stage.
+ */
+class RuntimeOptimizer : public ConstPass {
+public:
+    RuntimeOptimizer() = default;
+    RuntimeOptimizer(RuntimeConfigurator* configurator) : m_configurator(configurator) {}
+protected:
+    RuntimeConfigurator* m_configurator = nullptr;
```

Comment:
Discussed offline: the pointer is needed since CPU specific optimizers need `CPURuntimeConfigurator`. It was agreed to make this pointer `const` and add an assert in the constructor.